From 1cd42e0bce7a9cf70e1a0130bd1e81dfd9d8dbbc Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Fri, 16 Dec 2016 15:15:06 +0800
Subject: [PATCH 1360/5242] MLK-13625: ARM: dts: support spdif with imx51 evk
 expansion card

commit  a4f26977e827ac8240b98f0260a0a0ca61df52fd from
https://source.codeaurora.org/external/imx/linux-imx.git

The HDMI daughter card with sii902x will not be supported in
imx6sll-lpddr3-arm2 board, so use imx51 evk expansion card as spdif
interface

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6sll-lpddr3-arm2-spdif.dts |    6 ++++++
 arch/arm/boot/dts/imx6sll-lpddr3-arm2.dts       |   11 ++++++++---
 2 files changed, 14 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/imx6sll-lpddr3-arm2-spdif.dts b/arch/arm/boot/dts/imx6sll-lpddr3-arm2-spdif.dts
index 39f67da..5bcbc01 100644
--- a/arch/arm/boot/dts/imx6sll-lpddr3-arm2-spdif.dts
+++ b/arch/arm/boot/dts/imx6sll-lpddr3-arm2-spdif.dts
@@ -19,6 +19,12 @@
 	};
 };
 
+
+&iomuxc {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hog>;
+};
+
 &spdif {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_spdif>;
diff --git a/arch/arm/boot/dts/imx6sll-lpddr3-arm2.dts b/arch/arm/boot/dts/imx6sll-lpddr3-arm2.dts
index c524816..17381f7 100644
--- a/arch/arm/boot/dts/imx6sll-lpddr3-arm2.dts
+++ b/arch/arm/boot/dts/imx6sll-lpddr3-arm2.dts
@@ -398,7 +398,7 @@
 
 &iomuxc {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_hog>;
+	pinctrl-0 = <&pinctrl_hog &pinctrl_hog_sd2_reset>;
 
 	imx6sll-lpddr3-arm2 {
 		pinctrl_hog: hoggrp {
@@ -406,7 +406,6 @@
 				MX6SLL_PAD_KEY_ROW7__GPIO4_IO07 0x17059
 				MX6SLL_PAD_GPIO4_IO22__GPIO4_IO22 0x17059
 				MX6SLL_PAD_KEY_COL3__GPIO3_IO30	0x17059
-				MX6SLL_PAD_SD2_RESET__GPIO4_IO27 0x417059
 				MX6SLL_PAD_KEY_COL4__GPIO4_IO00 0x17059
 				MX6SLL_PAD_REF_CLK_32K__GPIO3_IO22 0x17059 /* SD3 CD */
 				MX6SLL_PAD_KEY_COL6__GPIO4_IO04 0x17059 /*SD3 RESET */
@@ -415,6 +414,12 @@
 			>;
 		};
 
+		pinctrl_hog_sd2_reset: hoggrp-1 {
+			fsl,pins = <
+				MX6SLL_PAD_SD2_RESET__GPIO4_IO27 0x417059
+			>;
+		};
+
 		pinctrl_audmux3: audmux3grp {
 			fsl,pins = <
 				MX6SLL_PAD_AUD_TXC__AUD3_TXC		0x4130b0
@@ -532,7 +537,7 @@
 
 		pinctrl_spdif: spdifgrp {
 			fsl,pins = <
-				MX6SLL_PAD_SD2_DATA4__SPDIF_OUT 0x4130b0
+				MX6SLL_PAD_SD2_RESET__SPDIF_OUT 0x4130b0
 			>;
 		};
 
-- 
1.7.9.5

